docs/learn/storage/block-sync.md

```diff
@@ -0,0 +1,73 @@
+
+#### Configuration
+
+- Archive mode: Persist every block states. Specify `--gcmode archive` flag.
+- Full mode: Persist some block states. Specify `--gcmode full` flag or simply omit the `--gcmode` flag.
+
+A node that runs on gcmode=archive is called Archive node. A node that runs gcmode=full is called Full node. Not to confuse with the full sync mode.
+
+In a full node, block states are persisted to disk every multiple of the number specified by `--state.block-interval NNN (default: 128)`. Also the block states of the recent `--state.tries-in-memory NNN (default 128)` blocks are kept in memory to serve APIs. Therefore, block states are only available when it is a multiple of block interval or recently processed.
+
+```js
+// State available
+> eth.getBalance('0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266', 150000000)
+735000000000002
+
+// State absent
+> eth.getBalance('0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266', 150000001)
+Error: missing trie node 64380a8de7bd83a6421c9ad45ae596a0eebbc7b504d061f4a57c61742eadc804 (path )
+	at web3.js:6812:9(39)
+	at send (web3.js:5223:62(29))
+	at <eval>:1:15(4)
+```
+
+#### Usage
+
+It's not a good idea for an application to assume block interval is 128. You might be tempted to rely on the 128 interval because 128 is the default value and most nodes use the default value. But it is the node’s discretion to choose any different interval.
+
+Applications usually need the latest states including nonces, balances, contract storages. Apps and developers sometime uses the debug tracing APIs for historic blocks, in which case the block can be recreated by re-executing from the nearest stored state (i.e. up to 127 blocks under default block-interval). So full node is an economic choice for general applications.
+
+However, you should use an archive node for data analysis purposes. Note that even if you only query historic consensus data - such as validators and rewards - you still need an archive node because they are derived from block states.
+
+#### Upstream EN
+
+If your node mostly serves the latest data but occasionally serves historic data, then try the Upstream EN feature. Read more [here](../upstream-en).
+
+### Live pruning
+
+Historic block states can be deleted to keep the disk size slim. With the live pruning feature enabled, the block states beyond a certain retention period will be automatically deleted. Read more about live pruning [here](../live-pruning).
+
+### Batch pruning (state migration)
+
+The block states can be state-migrated, meaning the block states before a certain block number is not available. Read more about state migration [here](../state-migration).
+
+## Chaindata snapshot
+
+Download a chaindata snapshot to skip the full sync process and start quickly. A chaindata snapshot is a compressed chaindata directory (e.g. `.tar.gz` file) of an already synced node. Read more about using the chaindata snapshot [here](../)
```

Comment:
Second sentence of `Snap sync` is more suitable here. Can we move the sentence from `Snap sync` to `Chaindata snapshot`?